src/cplus_plugin/gui/ncs_pwl_impact_manager_dialog.py

```diff
@@ -0,0 +1,544 @@
+        :param logicalIndex: Logical index of the section.
+        """
+        painter.save()
+        # Move the painter origin to the bottom-left of the section
+        painter.translate(rect.x() + rect.width(), rect.y())
+        painter.rotate(90)
+        # Paint the section using the base class at the new orientation
+        new_rect = QtCore.QRect(0, 0, rect.height(), rect.width())
+        super().paintSection(painter, new_rect, logicalIndex)
+        painter.restore()
+
+    def minimumSizeHint(self):
+        """
+        Returns the minimum size hint with transposed dimensions for rotation.
+        """
+        size = super().minimumSizeHint()
+        size.transpose()
+        return size
+
+    def sectionSizeFromContents(self, logicalIndex):
+        """
+        Returns the section size from contents, transposed for rotation.
+
+        :param logicalIndex: Logical index of the section.
+        """
+        size = super().sectionSizeFromContents(logicalIndex)
+        size.transpose()
+        return size
+
+
+class VerticalLabel(QtWidgets.QLabel):
+    """
+    A QLabel subclass that displays its text vertically (rotated 270 degrees).
+    Used for vertical header in the matrix table.
+    """
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+    def paintEvent(self, event):
+        """
+        Paints the label text rotated 270 degrees (vertical orientation).
+        """
+        painter = QtGui.QPainter(self)
+        painter.translate(0, self.height())
+        painter.rotate(-90)
+        painter.drawText(
+            0, 0, self.height(), self.width(), QtCore.Qt.AlignCenter, self.text()
+        )
+
+    def sizeHint(self):
+        """
+        Returns the size hint, swapping width and height for vertical orientation.
+        """
+        size = super().sizeHint()
+        return size.transposed()
+
+
+class TransMatrixEdit(QtWidgets.QLineEdit):
+    """
+    Custom QLineEdit for matrix cell editing with validation and color feedback.
+    """
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.editingFinished.connect(self.style_cell)
+        self.textChanged.connect(self.validate_cell)
+
+    def validate_cell(self):
+        """Validate the cell input. Values must be integers between -3 and 3, or empty."""
+        text = self.text().strip()
+        if text == "" or text == "-":
+            # Allow empty input
+            return
+        try:
+            value = int(text)
+            if value < -3 or value > 3:
+                raise ValueError
+        except ValueError:
+            # Invalid input: reset to empty
+            self.setText("")
+
+    def style_cell(self):
+        """
+        Validates the cell input and updates the background color accordingly.
+        """
+        text = self.text().strip()
+        # Reset style for empty input
+        if text == "":
+            self.setText("")
+            self.setStyleSheet(
```

Comment:
This is repeated 4 times. Rather make a constant at the top/beginning of the file lile
`DEFAULT_CELL_STYLE = "QLineEdit {background: white;} QLineEdit:hover {border: 1px solid gray; background: white;}"`
Then `self.setStyleSheet(DEFAULT_CELL_STYLE`